### PR TITLE
faker: register all possible transformers

### PR DESCRIPTION
**Pull request by beuss**
Instead of hardcoding a very short list of transformers, simply use
reflexion to fetch all available ones.

### CODE REVIEW

**Comment by beuss** [comment]
Once merged this one will follow https://github.com/beuss/gdpr-dump/commit/f727075fd6fdcdf7f38edefb05d4699afef38615

**Comment by bomoko** [comment]
Hi @beuss 

Apologies for taking so long to get to these - I'll review them (and merge) by the end of the week.

**Comment by beuss** [comment]
@bomoko no problem, having a life outside github is a sane thing :)

File: src/ColumnTransformer/Plugins/FakerColumnTransformer.php [hunk 1]
```diff
@@ -33,13 +28,23 @@ protected function getSupportedFormatters()
 
     public function __construct()
     {
-        if (!isset(self::$factory)) {
-            self::$factory = Factory::create();
+        if (!isset(self::$generator)) {
+            self::$generator = Factory::create();
+            foreach(self::$generator->getProviders() as $provider)
+            {
```
Comment (segadora) [inline src/ColumnTransformer/Plugins/FakerColumnTransformer.php]: This bracket should be moved to the line above

File: src/ColumnTransformer/Plugins/FakerColumnTransformer.php [hunk 2]
```diff
@@ -33,13 +28,23 @@ protected function getSupportedFormatters()
...
+                $clazz = new \ReflectionClass($provider);
+                $methods = $clazz->getMethods(\ReflectionMethod::IS_PUBLIC);
+                foreach($methods as $m)
```
Comment (segadora) [inline src/ColumnTransformer/Plugins/FakerColumnTransformer.php]: This bracket should be moved to the line above

File: src/ColumnTransformer/Plugins/FakerColumnTransformer.php [hunk 3]
```diff
@@ -33,13 +28,23 @@ protected function getSupportedFormatters()
...
+                {
+                    if(strpos($m->name, '__') === 0) continue;
```
Comment (segadora) [inline src/ColumnTransformer/Plugins/FakerColumnTransformer.php]: Missing space after if

**Comment by bomoko** [comment]
Apologies for taking so long on this again @beuss - I was trying to dig up help internally for some design decisions, alas, nothing!
Going through the rest of the PRs now

**Pull request merged**